coverage_assessment.R

```diff
@@ -41,9 +41,131 @@ nut <- get_data()
 
 # Problem 3: Prevalence of acute undernutrition/malnutrition ----
 
+####MAM
 
+with(nut, sum(muac >= 11.5 & muac <12.5  & age >=6 & age <60 & oedema == 2, na.rm= TRUE))
 
+##number of moderate acute malnourished (MAM) cases children 6-59 months old 
 
+Male_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 & age >=6 & age <60 & oedema == 2 & sex == 1 , na.rm= TRUE ))
+
+##assigning moderate acute malnourished (MAM) male 6-59 months old to Male_MAM
+
+Female_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 & age >=6 & age <60 & oedema == 2 & sex == 2 , na.rm= TRUE ))
+
+##assigning  moderate acute malnourished (MAM) female 6-59 months old to Female_MAM
+
+UM_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 & age >=6 & age <60 & oedema == 2 & county == "Urban Montserrado" , na.rm= TRUE))
+
+##assigning moderate acute malnourished (MAM) cases children 6-59 months old  in Urban Montserrado to UM_MAM
+
+GB_MAM <- with(nut, sum(muac >= 11.5 & muac <12.5 &age >=6 & age <60 & oedema == 2 & county == "Grand Bassa" , na.rm= TRUE))
+
+##assigning  moderate acute malnourished (MAM) cases children 6-59 months old other than Urban Montserrado to GB_MAM
+
+MAM <- data.frame(Male_MAM, Female_MAM, UM_MAM, GB_MAM)
+
+## creating a data frame for MAM
+
+if(interactive()) View(MAM)
+
+##Viewing MAM data
+
+
+
+####Alternate table method for number of moderate,severe and normal malnutrition by sex and by location:
+
+num_data <- with(nut, data.frame(sex,county,muac,oedema))
+
+##assigning the data frame to num_data
+
+SEX <- ifelse(num_data$sex == "1", "Male", "Female")
+OEDEMA <- ifelse(num_data$oedema == "1", "Yes", "No")
+MUAC <- with( num_data, ifelse (muac <11.5, "Severe",
+                               ifelse(oedema == 1, "Severe",
+                                      ifelse(muac >=11.5 & muac<12.5, "Moderate",
+                                             ifelse (muac >=12.5, "Normal", "No")))))
+
+list <- data.frame(SEX,MUAC, OEDEMA, num_data$county)
+
+##creating data frame by sex,muac,oedema and location
+
+
+tab <-table(SEX,num_data$county, MUAC )
+
+##creating table showing different numbers of children according to moderate,severe degree of Malnutrition and normal children
+
+tab
```

Comment:
This is a good attempt at an alternative approach. My comments are:

* in line 78, the syntax is correct but you didn't really have to make a new data.frame just for those values. You can still do what you did in lines 82 to 84 without having to create `num_data`. You could have just done this:

```R
## Recode sex
SEX <- ifelse(nut$sex == 1, "Male", "Female")    ## Note here that I use 1 instead of "1" for consistency and because the sex variable is an integer

## Recode oedema
OEDEMA <- ifelse(nut$oedema == 1, "Yes", "No")                                 ## Again, here I use 1 and not "1"

## Recode MUAC, oedema, and age to classify kids as SAM or not SAM
AM <- ifelse(nut$muac < 11.5 | nut$oedema == 1, "SAM", "not SAM")              ## Based on muac and oedema, classify child as SAM or not SAM

AM <- ifelse(nut$muac >= 11.5 & nut$muac < 12.5 & nut$oedema == 2, "MAM", AM)  ## Based on muac and oedema, classify child as SAM or MAM or not SAM

AM <- ifelse(AM == "not SAM", "not SAM or MAM", AM)                            ## Recode not SAM to not SAM or MAM

AM <- ifelse(nut$age >= 6 & nut$age < 60, AM, NA)  ## Based on age, classify whether child is eligible for the analysis

## Add the new variables to the nut object - instead of creating a new object
nut$SEX <- SEX
nut$OEDEMA <- OEDEMA
nut$AM <- AM

```